src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp

```diff
@@ -509,6 +488,53 @@ void PrettyBlockOutputFormat::writeValueWithPadding(
     }
 }
 
+void PrettyBlockOutputFormat::writeHeaderWithPadding(
+    const Widths & max_widths, const Widths & name_widths, const char * separator, size_t col_num, const ColumnWithTypeAndName & col)
+{
+    String visible_col_name = col.name;
+    UInt64 name_width = name_widths[col_num];
+    if (name_width > max_widths[col_num])
+    {
+        const char * ellipsis = format_settings.pretty.charset == FormatSettings::Pretty::Charset::UTF8 ? "⋯" : "~";
+        const size_t front_bytes = UTF8::computeBytesBeforeWidth(
+            reinterpret_cast<const UInt8 *>(col.name.data()),
+            col.name.size(),
+            0,
+            static_cast<size_t>(std::ceil((max_widths[col_num] - 1) / 2.0)));
+
```

Comment:
Should we use `format_settings.pretty.max_column_name_width` instead of `max_widths[col_num]` which make truncation of column names width less than or equal with `max_column_name_width`?

Reply:
Actually, it is not max_column_name_width, but max_column_pad_width, AFAIR.